Source/Filter/BilinearFilter.cpp

```diff
@@ -64,34 +64,25 @@ void BilinearFilter::ComputeStencils(){
     WARPX_PROFILE("BilinearFilter::ComputeStencils()");
     int i = 0;
     for (const auto& el : npass_each_dir ) {
-        stencil_length_each_dir[i++] = static_cast<int>(el);
+        stencil_length_each_dir[i++] = static_cast<int>(el) + 1;
     }
-    stencil_length_each_dir += 1.;
-#if defined(WARPX_DIM_3D)
-    // npass_each_dir = npass_x npass_y npass_z
+
     stencil_x.resize( 1u + npass_each_dir[0] );
-    stencil_y.resize( 1u + npass_each_dir[1] );
-    stencil_z.resize( 1u + npass_each_dir[2] );
     compute_stencil(stencil_x, npass_each_dir[0]);
+#if AMREX_SPACEDIM >= 2
+    stencil_y.resize( 1u + npass_each_dir[1] );
     compute_stencil(stencil_y, npass_each_dir[1]);
+#endif
+#if AMREX_SPACEDIM == 3
+    stencil_z.resize( 1u + npass_each_dir[2] );
     compute_stencil(stencil_z, npass_each_dir[2]);
-#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-    // npass_each_dir = npass_x npass_z
-    stencil_x.resize( 1u + npass_each_dir[0] );
-    stencil_z.resize( 1u + npass_each_dir[1] );
-    compute_stencil(stencil_x, npass_each_dir[0]);
-    compute_stencil(stencil_z, npass_each_dir[1]);
-#elif defined(WARPX_DIM_1D_Z)
-    // npass_each_dir = npass_z
-    stencil_z.resize( 1u + npass_each_dir[0] );
-    compute_stencil(stencil_z, npass_each_dir[0]);
 #endif
+
     slen = stencil_length_each_dir.dim3();
-#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+#if AMREX_SPACEDIM < 3
     slen.z = 1;
 #endif
-#if defined(WARPX_DIM_1D_Z)
+#if AMREX_SPACEDIM < 2
```

Comment:
For the logic here, can we express this also as `or`-concatenated branches of the `WARPX_` macros?

For clarity reasons (e.g., is RZ truly 2D or something else), we factored out all (most?) use of the `AMREX_SPACEDIM` macro in WarpX a while ago :)

Reply:
Yes, I can use the `WARPX_DIM` macros. But I was thinking of this differently. The code is generic to the geometry - it is only the spatial dimensions of the grid that matter. So RZ and XZ are both 2D. I thought it was nice to have the code written without the `WARPX_DIM` macros emphasizing that the code is only dependent on the number of dimensions (and particularly, does not depend on the ordering of dimensions).